staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_discovery.go

```diff
@@ -406,14 +406,38 @@ func (dm *discoveryManager) syncAPIService(apiServiceName string) error {
 	return nil
 }
 
+func (dm *discoveryManager) markInitialStale() {
+	dm.servicesLock.Lock()
+	defer dm.servicesLock.Unlock()
+
+	for key, info := range dm.apiServices {
```

Comment:
Is the intention to also add local services, or we are not worried since they will definitely sync? Asking because IIUC `dm.apiServices` only includes services whose `Spec.Service` is non-null

Reply:
Local apiservices should be guaranteed to sync, but as you mentioned in the linked bug they have their own readiness block where they would be 100% available before health check succeeds. I think it's fine to just look at  non-local apiservices?

Reply:
makes sense